### ISSUE TITLE
RoleBinding is generated in operator namespace instead of watched namespaces

### ISSUE THREAD

**Issue by tomrss** [quarkiverse/quarkus-operator-sdk#447]
When setting Kubernetes namespace, the operator seems to be limited to watch custom resources in that namespace. 
Generated Kubernetes YAML has the ClusterRole for the reconciler bound with RoleBinding in current (operator) namespace:
`application.properties`:
```
quarkus.kubernetes.namespace=myoperator-namespace
quarkus.operator-sdk.controllers."mycrcontroller".namespaces=mycr-namespace
```
Generated `target/kubernetes/kubernetes.yml`:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: mycrreconciler-cluster-role
  namespace: myoperator-namespace
rules:
  - apiGroups:
      - my.group
    resources:
      - mycr
      - mycr/status
      - mycr/finalizers
    verbs:
      - get
      - list
      - watch
      - create
      - delete
      - patch
      - update
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: mycrreconciler-role-binding
  namespace: myoperator-namespace
roleRef:
  kind: ClusterRole
  apiGroup: rbac.authorization.k8s.io
  name: mycrreconciler-cluster-role
subjects:
  - kind: ServiceAccount
    name: mycr-operator
    namespace: myoperator-namespace
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: mycr-operator-view
  namespace: myoperator-namespace
roleRef:
  kind: ClusterRole
  apiGroup: rbac.authorization.k8s.io
  name: view
subjects:
  - kind: ServiceAccount
    name: mycr-operator
    namespace: myoperator-namespace
---
# other...
```
If I deploy with configuration above, operator can't start:
```
2022-11-24 10:37:54,274 ERROR [io.jav.ope.Operator] (main) Error starting operator: io.javaoperatorsdk.operator.OperatorException: Couldn't start source ControllerResourceEventSource
	at io.javaoperatorsdk.operator.processing.event.EventSourceManager.startEventSource(EventSourceManager.java:97)
	at io.javaoperatorsdk.operator.processing.event.EventSourceManager.start(EventSourceManager.java:64)
	at io.javaoperatorsdk.operator.processing.Controller.start(Controller.java:295)
	at io.javaoperatorsdk.operator.ControllerManager.lambda$start$0(ControllerManager.java:35)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1779)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:686)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)
	at io.javaoperatorsdk.operator.ControllerManager.start(ControllerManager.java:35)
	at io.javaoperatorsdk.operator.Operator.start(Operator.java:101)
	at io.quarkiverse.operatorsdk.runtime.AppEventListener.onStartup(AppEventListener.java:28)
	at io.quarkiverse.operatorsdk.runtime.AppEventListener_Observer_onStartup_9c5d3dd7eff7dbe02976f59f48aec5b70eb9c99f.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:326)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:308)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:76)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:131)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:100)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:110)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:69)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:122)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://10.0.0.1/apis/my.group/v1/namespaces/mycr-namespace/mycrs. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. mycrs.my.group is forbidden: User "system:serviceaccount:myoperator-namespace:mycr-operator" cannot list resource "mycr" in API group "my.group" in the namespace "mycr-namespace".
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:684)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:664)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:613)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:558)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:521)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:505)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:133)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:415)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:83)
	at io.fabric8.kubernetes.client.informers.cache.Reflector.listSyncAndWatch(Reflector.java:96)
	at io.fabric8.kubernetes.client.informers.cache.Reflector.start(Reflector.java:75)
	at io.fabric8.kubernetes.client.informers.impl.DefaultSharedIndexInformer.run(DefaultSharedIndexInformer.java:152)
	at io.javaoperatorsdk.operator.processing.event.source.informer.InformerWrapper.start(InformerWrapper.java:35)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3612)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:686)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)
	at io.javaoperatorsdk.operator.processing.event.source.informer.InformerManager.start(InformerManager.java:43)
	at io.javaoperatorsdk.operator.processing.event.source.informer.ManagedInformerEventSource.start(ManagedInformerEventSource.java:70)
	at io.javaoperatorsdk.operator.processing.event.source.controller.ControllerResourceEventSource.start(ControllerResourceEventSource.java:61)
	at io.javaoperatorsdk.operator.processing.event.NamedEventSource.start(NamedEventSource.java:20)
	at io.javaoperatorsdk.operator.processing.event.EventSourceManager.startEventSource(EventSourceManager.java:92)
	... 39 more
```

Is this the expected behaviour, or maybe I missed something in configuration? I feel like it is common usage to keep the operator in a different namespace then custom resources.

Version used: 4.0.5

Thanks,
Tommaso

**Comment by metacosm** [comment]
The configuration generation doesn't indeed take the watched namespaces into account at the moment. The reason for this is that the value could be overridden at runtime, thus rendering using it at build time, moot. However, we could indeed make the effort of generating something that is correct based on the build time available information, with the understanding that it might become invalid if some of that information is changed at runtime.

**Comment by tomrss** [comment]
Thank you, I understand the problem. Do you think that is feasible to use a ClusterRoleBinding instead of RoleBinding, in order to be deployable right away?

**Comment by metacosm** [comment]
Hi @tomrss, could you check if #456 fixes your issue, please?

**Comment by metacosm** [comment]
Fixed by #456.

**Comment by tomrss** [comment]
Hi @metacosm, I confirm that is fixed.
Thank you!
